source/fem-dem/cfd_dem_coupling.cc

```diff
@@ -742,6 +1097,10 @@ CFDDEMSolver<dim>::dem_contact_build(unsigned int counter)
 
       // Particles-wall fine search
       particle_wall_fine_search();
+
+      // Reset different steps
+      checkpoint_step   = false;
+      load_balance_step = false;
```

Comment:
You need to document a bit why you have this reset mechanism like this. Otherwise it is difficult to understand.

Reply:
done